vendor/k8s.io/kube-openapi/pkg/builder/openapi.go

```diff
@@ -45,7 +45,46 @@ type openAPI struct {
 
 // BuildOpenAPISpec builds OpenAPI spec given a list of webservices (containing routes) and common.Config to customize it.
 func BuildOpenAPISpec(webServices []*restful.WebService, config *common.Config) (*spec.Swagger, error) {
-	o := openAPI{
+	o := newOpenAPI(config)
+
+	err := o.init()
+	if err != nil {
+		return nil, err
+	}
+	err = o.buildPaths(webServices)
+	if err != nil {
+		return nil, err
+	}
+	err = o.finishSpec()
+	if err != nil {
+		return nil, err
+	}
+
+	return o.swagger, nil
+}
+
+// BuildOpenAPISpecForResource builds a partial OpenAPI spec given a sample object and common.Config to customize it.
+func BuildOpenAPISpecForResource(model interface{}, config *common.Config) (*spec.Swagger, error) {
+	o := newOpenAPI(config)
+
+	err := o.init()
+	if err != nil {
+		return nil, err
+	}
+	_, err = o.toSchema(model)
```

Comment:
what does toSchema do? Seems odd to discard the result

Reply:
I could try to add a comment above the line explaining it in more detail